examples/o1/Using_chained_calls_for_o1_structured_outputs.ipynb

```diff
@@ -0,0 +1,266 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "companies=[CompanyData(company_name='Walmart', page_link='https://en.wikipedia.org/wiki/Walmart', reason='As the largest retailer, Walmart can leverage AI for supply chain optimization, inventory management, personalized customer experiences, and operational efficiency.'), CompanyData(company_name='ExxonMobil', page_link='https://en.wikipedia.org/wiki/ExxonMobil', reason='ExxonMobil can benefit from AI in predictive maintenance, optimizing exploration and drilling, analyzing geological data, and enhancing safety measures.'), CompanyData(company_name='UnitedHealth Group', page_link='https://en.wikipedia.org/wiki/UnitedHealth_Group', reason='UnitedHealth Group can utilize AI to improve patient data analysis, diagnostics, personalize treatment plans, detect fraud, and increase operational efficiency.')]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import requests\n",
+    "from pydantic import BaseModel\n",
+    "from openai import OpenAI\n",
+    "\n",
+    "client = OpenAI()\n",
+    "\n",
+    "def fetch_html(url):\n",
+    "    response = requests.get(url)\n",
+    "    if response.status_code == 200:\n",
+    "        return response.text\n",
+    "    else:\n",
+    "        return None\n",
+    "\n",
+    "url = \"https://en.wikipedia.org/wiki/List_of_largest_companies_in_the_United_States_by_revenue\"\n",
+    "html_content = fetch_html(url)\n",
+    "\n",
+    "class CompanyData(BaseModel):\n",
+    "    company_name: str\n",
+    "    page_link: str\n",
+    "    reason: str\n",
+    "\n",
+    "class CompaniesData(BaseModel):\n",
+    "    companies: list[CompanyData]\n",
+    "\n",
+    "json_format = \"\"\"\n",
+    "{\n",
+    "    \\\"company_name\\\": \\\"OpenAI\\\",\n",
+    "    \\\"page_link\\\": \\\"https://en.wikipedia.org/wiki/OpenAI\\\",\n",
+    "    \\\"reason\\\": \\\"OpenAI would benefit because they are an AI company...\\\"}\"\n",
+    "\"\"\"\n",
+    "\n",
+    "o1_response = client.chat.completions.create(\n",
+    "    model=\"o1-preview\",\n",
+    "    messages=[\n",
+    "        {\n",
+    "            \"role\": \"user\", \n",
+    "            \"content\": f\"\"\"\n",
+    "You are a business analyst designed to understand how AI technology could be used across large corporations.\n",
+    "\n",
+    "Read the following html and return which companies would benefit from using AI technology: {html_content}. Rank these propects by opportunity by comparing them and show me the top 3. Return each company with the following information: {CompanyData.model_fields.keys()}\"\n",
+    "\"\"\"\n",
+    "        }\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "o1_response_content = o1_response.choices[0].message.content\n",
+    "\n",
+    "response = client.beta.chat.completions.parse(\n",
+    "    model=\"gpt-4o-2024-08-06\",\n",
```

Comment:
Would also recommend using 4o-mini for this example. Since this is a relatively simple task, it would be good to showcase the smallest/cheapest model that can do the job, especially if they are pairing this with an expensive model like o1-preview

Reply:
Ah yeah, sounds great!